src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4754,6 +4754,14 @@ private unsafe char[] GetCanonicalPath(char[] dest, ref int pos, UriFormat forma
                         end += (_info.Offset.Query - _info.Offset.Path);
                     }
                 }
+
+                // On Unix, escape '\\' in path of file uris to '%5C' canonical form.
+                if (!IsWindowsSystem && _syntax.NotAny(UriSyntaxFlags.ConvertPathSlashes) && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && !IsImplicitFile)
```

Comment:
> We'd prefer if the code doesn't exist at all in Windows builds to ensure app-compat in the case where the Uri has been serialized and stored to disk in any form.

This code will not exist in windows version of the dll, the compiler will realize this is unreachable codepath, and not include this if block in the IL. 

```c#
using System;

namespace TestBoolean
{
	public class Program
	{
		private const bool answer = true;
		public static void Main(string[] args)
		{
			if (!answer)
				Console.WriteLine("Hello World");
		}
	}
}
```

will give warning, unreachable code detected, and the IL does not have the if branch,

```il
.class public auto ansi beforefieldinit TestBoolean.Program
	extends [mscorlib]System.Object
{
	// Fields
	.field private static literal bool answer = bool(true)

	// Methods
	.method public hidebysig static 
		void Main (
			string[] args
		) cil managed 
	{
		// Method begins at RVA 0x2050
		// Code size 6 (0x6)
		.maxstack 1
		.entrypoint
		.locals init (
			[0] bool
		)

		IL_0000: nop
		IL_0001: ldc.i4.0
		IL_0002: stloc.0
		IL_0003: br.s IL_0005

		IL_0005: ret
	} // end of method Program::Main

	.method public hidebysig specialname rtspecialname 
		instance void .ctor () cil managed 
	{
		// Method begins at RVA 0x2062
		// Code size 8 (0x8)
		.maxstack 8

		IL_0000: ldarg.0
		IL_0001: call instance void [mscorlib]System.Object::.ctor()
		IL_0006: nop
		IL_0007: ret
	} // end of method Program::.ctor

} // end of class TestBoolean.Program
```